Src/ILGPU/Backends/SPIRV/SPIRVBuilderUtils.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace ILGPU.Backends.SPIRV
+{
+    /// <summary>
+    /// Utilities for generated functions in SPIRVBuilder.cs
+    /// </summary>
+    public static class SPIRVBuilderUtils
+    {
+        /// <summary>
+        /// Attempts to convert an object to a SPIRV compatible representation
+        /// </summary>
+        /// <param name="obj">The object to convert</param>
+        /// <returns>A list of words (uints) representing the object</returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public static List<uint> ToUintList(object obj)
```

Comment:
I am not sure about this function in general, as it seems to be dangerous to use. Can we split/refine this function to avoid using objects?

Reply:
I agree. This function is not very cohesive and prone to error. I'll split it up to be more type safe.

Question: There will still be some amount of danger as I don't see a way to get around reflection. Is this okay? Can you think of a way to get around it?